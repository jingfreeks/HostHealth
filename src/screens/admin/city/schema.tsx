import * as yup from 'yup';
import {message} from '@/config/constant'
export const Schema = yup
  .object({
    name: yup.string().required('City Name Should be required'),
    state: yup.string().required('State Name Should be required')
  })
  .required();
