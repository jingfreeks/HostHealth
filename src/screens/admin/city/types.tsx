import { RouteProp } from '@react-navigation/native';
import type {AppNavigationProps} from '@/navigation/types';
type StateFormScreenRouteProp = RouteProp<AppNavigationProps, 'Cityform'>;
export type RoutesProps = {
  route: StateFormScreenRouteProp
};

export type StateFormProps = {
  name:string;
  state:string;
  _id:string;
};
