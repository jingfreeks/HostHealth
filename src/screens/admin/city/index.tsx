export {default as City} from './city'
export {default as Cityform} from './form' 