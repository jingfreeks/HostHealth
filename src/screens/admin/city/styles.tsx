import styled from 'styled-components/native';
import {verticalScale} from 'react-native-size-matters';
import {Text} from '@/component';

export const ListItemContainerStyled = styled.View`
  align-items: center;
  padding: 10px;
`;

export const ContainerStyled = styled.SafeAreaView`
  flex: 1;
  margin-bottom: 10px;
`;

export const FormContainerStyled = styled.View`
  flex: 1;
  justify-content: center;
  margin-left: ${verticalScale(20)}px;
  margin-right: ${verticalScale(20)}px;
`;
export const FormHeaderContainerStyled = styled.View`
  align-items: center;
  margin-bottom: ${verticalScale(20)}px;
`;
export const FormTextInputContainerStyled = styled.View`
  margin-top: ${verticalScale(20)}px;
  margin-bottom: ${verticalScale(5)}px;
`;

export const FormHeaderTextStyled = styled(Text)`
  font-size: 30px;
  font-weight: 900;
`;

export const StateEmptyContainerStyled=styled.View`
  flex:1;
  align-items:center;
  justify-content:center;
`
export const ProfileAvatarContainerStyled = styled.View`
  align-items: center;
  justify-content: center;
  margin-top: 20px;
`;