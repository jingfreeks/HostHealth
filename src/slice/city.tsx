/* eslint-disable @typescript-eslint/no-explicit-any */
import { createEntityAdapter } from '@reduxjs/toolkit';
import {apiSlice} from '@/config/apiSlice';

type cityTypes = { _id: string; name: string }

const cityAdapter=createEntityAdapter<cityTypes>({
    selectId:(city)=>city._id,
    sortComparer:(a,b)=>a.name.localeCompare(b.name)
})
const initialState = cityAdapter.getInitialState()
export const cityApiSlice = apiSlice.injectEndpoints({
  endpoints: builder => ({
    getCity: builder.query({  
        query: () => '/city',
        transformResponse: responseData => {
            // let min = 1;
            // const loadedPosts = responseData.map(post => {
            //     if (!post?.date) post.date = sub(new Date(), { minutes: min++ }).toISOString();
            //     if (!post?.reactions) {post.reactions = {
            //         thumbsUp: 0,
            //         wow: 0,
            //         heart: 0,
            //         rocket: 0,
            //         coffee: 0
            //     }}
            //     return post;
            // });
            return cityAdapter.setAll(initialState, responseData)
        },
        providesTags: (result:any, error, arg):any => [
            { type: 'City', id: 'LIST' },
            ...result.ids.map((id:string| number) => ({ type: 'City', id }))
        ]
    }),
  }),
  overrideExisting: false,
});

export const {useGetCityQuery} = cityApiSlice;
