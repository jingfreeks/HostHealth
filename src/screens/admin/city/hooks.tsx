import React, {useState,useCallback} from 'react';
import {Alert} from 'react-native';
import {StackNavigationProp} from '@react-navigation/stack';
import type {RootNavigationProps} from '@/navigation/types';
import {useNavigation} from '@react-navigation/native';
import {launchImageLibrary} from 'react-native-image-picker';
import {useGetCityQuery, useDeleteCityMutation,useUploadProfileMutation,useAddCityMutation,useUpdateCityMutation} from '@/slice';
import {useForm,SubmitHandler} from 'react-hook-form';
import {yupResolver} from '@hookform/resolvers/yup';
import {Schema} from './schema';
import * as yup from 'yup';
import Config from 'react-native-config';

export const useCityHooks = () => {
  const navigation = useNavigation<StackNavigationProp<RootNavigationProps>>();
  const [uri, setUri] = useState<string>('');

  type FormData = yup.InferType<typeof Schema>;
  const formMethod = useForm<FormData>({
    defaultValues: {
      name: '',
      state:'',
      cityImage:'',
      id:'',
    },
    resolver: yupResolver(Schema),
  });

  const {
    data: city,
    refetch: cityRefresh,
    isLoading,
    isSuccess,
    isError,
    error,
  } = useGetCityQuery<{
    refetch: () => void;
    data: any;
    isLoading: boolean;
    isSuccess: string;
    isError: boolean;
    error: any;
  }>('getCity');

  const options: {
    saveToPhotos: boolean;
    mediaType: 'photo' | 'video';
    includeBase64: boolean;
  } = {
    saveToPhotos: true,
    mediaType: 'photo',
    includeBase64: false,
    // includeExtra,
  };
  const [addCity, {isLoading: addStateLoading}] = useAddCityMutation();
  const [updateCity, {isLoading: updateStateLoading}] = useUpdateCityMutation();
  const [deleteCity, {isLoading: deleteLoading}] = useDeleteCityMutation();
  const handleDeleteCity = async (id: string) => {
    try {
      await deleteCity({
        id,
      }).unwrap();
    } catch (error) {
      Alert.alert(error?.data?.message);
    }
  };
  const onRefresh = async () => {
    await cityRefresh();
  };
  const [uploadProfile] = useUploadProfileMutation();

  const onSubmit: SubmitHandler<FormData> = async data => {
    try {
      let response: any;
      if (data.id) {
        //update
        response = await updateCity({
          name: data?.name,
          stateId: data?.state,
          image:data.cityImage,
          id: data.id,
        }).unwrap();
      } else {
        //insert
        response = await addCity({
          name: data?.name,
          stateId: data?.state,
          image:data.cityImage,
        });
      }
      if (response?.error) {
        alert(response?.error?.data?.message);
      } else {
        navigation.goBack();
      }
    } catch (error) {
      console.log('error', error);
    }
  };

  const handleViewImage = useCallback(async () => {
    const result: any = await launchImageLibrary(options);
    if (result?.assets) {
      const response = await uploadProfile(result);
      const {data}: any = (await response) || {};
      formMethod.setValue('cityImage', `${Config.DEV_BASE_URL}${data?.url}`);
      setUri(`${Config.DEV_BASE_URL}/${data?.url}`);
    }
  }, [uri]);

  return {
    navigation,
    city,
    isLoading,
    cityLoading: isLoading,
    isSuccess,
    isError,
    error,
    handleDeleteCity,
    deleteLoading,
    onRefresh,
    uri,
    setUri,
    options,
    uploadProfile,
    addCity,
    updateCity,
    addStateLoading,
    updateStateLoading,
    formMethod,
    onSubmit,
    handleViewImage
  };
};
