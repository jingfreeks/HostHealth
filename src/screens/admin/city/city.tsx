import React, {useCallback} from 'react';
import {
  FlatList,
  ListRenderItem,
  ListRenderItemInfo,
} from 'react-native';
import {FAB} from 'react-native-elements';
import {ContainerStyled,StateEmptyContainerStyled} from './styles';
import {List} from './component';
import {HomeEmptyCard} from '@/component';
import {PcitiesEmptyIcon} from '@/assets';
import {useCityHooks} from './hooks';

const City = () => {
  let content;
  const {navigation, city, isLoading, isSuccess, error, isError} =
  useCityHooks();

  const renderItem: ListRenderItem<any> = useCallback(
    ({item}: ListRenderItemInfo<any>) => {
      return <List cityId={item} />;
    },
    [],
  );
  console.log('city',city)
  if (isLoading || isError) {
    let messages;
    if (error?.status === 403) {
      messages = 'Token Expired you must logout and login it again';
    } else {
      messages = 'No cities matches your preferred locations';
    }
    content = (
      <StateEmptyContainerStyled>
        <HomeEmptyCard imgsource={PcitiesEmptyIcon} message={messages} />
      </StateEmptyContainerStyled>
    );
  } else if (isSuccess) {
    content = (
      <ContainerStyled>
        <FlatList
          data={city.ids}
          extraData={city.ids}
          renderItem={renderItem}
          keyExtractor={(item: any, index) => index.toString()}
        />
        <FAB
          title="Create"
          placement="right"
          size="large"
          onPress={() => navigation.navigate('Cityform')}
        />
      </ContainerStyled>
    );
  }
  return content;
};

export default City;
