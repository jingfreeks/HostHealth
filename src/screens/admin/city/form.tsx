import React, {useState, useMemo} from 'react';
import {
  FormHeaderContainerStyled,
  FormTextInputContainerStyled,
  FormContainerStyled,
  FormHeaderTextStyled,
} from './styles';
import {StyleSheet} from 'react-native';
import {
  FormTextController,
  Formdropdowncontroller,
  Bbutton,
} from '@/component';
import {useCityHooks} from './hooks';
import {colors} from '@/utils/themes';
import {useForm, FormProvider, SubmitHandler,} from 'react-hook-form';
import {yupResolver} from '@hookform/resolvers/yup';
import {useAddCityMutation, useUpdateCityMutation} from '@/slice/city';
import {Schema} from './schema';
import * as yup from 'yup';
import type {RoutesProps} from './types';
import {useStateHooks} from '@/screens/admin/state/hooks';

const Form = (props: RoutesProps) => {
  const {navigation} = useCityHooks();

  const [isFocus, setIsFocus] = useState(false);
  const {route} = props;
  const {name, state, _id} = route?.params || {};
  // const [value, setValue] = useState<any>(state);
  type FormData = yup.InferType<typeof Schema>;
  const formMethod = useForm<FormData>({
    defaultValues: {
      name: name || '',
      state:state || '',
    },
    resolver: yupResolver(Schema),
  });

  const [addCity, {isLoading: addStateLoading}] = useAddCityMutation();
  const [updateCity, {isLoading: updateStateLoading}] = useUpdateCityMutation();

  const {states, isLoading: stateloading} = useStateHooks();

  const statesdata = useMemo(() => {
    let statedata = states;
    return states?.ids?.map((item: string) => {
      return {
        label: statedata?.entities[item]?.name,
        value: item,
      };
    });
  }, [states]);

  const onSubmit: SubmitHandler<FormData> = async data => {
    try {
      let response: any;
      if (_id) {
        //update
        response = await updateCity({
          name: data?.name,
          stateId: data?.state,
          image:
            'https://img.freepik.com/premium-photo/city-skyline-with-city-background_1249034-36162.jpg?w=826',
          id: _id,
        }).unwrap();
      } else {
        //insert
        response = await addCity({
          name: data?.name,
          stateId: data?.state,
          image:
            'https://img.freepik.com/premium-photo/city-skyline-with-city-background_1249034-36162.jpg?w=826',
        });
      }
      if (response?.error) {
        alert(response?.error?.data?.message);
      } else {
        navigation.goBack();
      }
    } catch (error) {
      console.log('error', error);
    }
  };
  return (
    <FormContainerStyled>
      <FormProvider {...formMethod}>
        <FormHeaderContainerStyled>
          <FormHeaderTextStyled TextMode="Htitlenormal">
            City Form Information
          </FormHeaderTextStyled>
        </FormHeaderContainerStyled>
        <FormTextInputContainerStyled>
          <FormTextController
            Label="Name"
            name="name"
            placeholder="Name"
            rules={{
              required: true,
            }}
          />
        </FormTextInputContainerStyled>

        <Formdropdowncontroller
          Label="State"
          name="state"
          placeholder="State"
          rules={{
            required: true,
          }}
          loading={stateloading}
          data={statesdata}
        />
        <Bbutton
          bcolor={colors.lightergreen}
          border={10}
          loaders={addStateLoading || updateStateLoading}
          title="Save"
          onPress={formMethod.handleSubmit(onSubmit)}
        />
      </FormProvider>
    </FormContainerStyled>
  );
};

export default Form;
const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    padding: 16,
  },
  dropdown: {
    height: 50,
    borderColor: 'gray',
    borderWidth: 0.5,
    borderRadius: 8,
    paddingHorizontal: 8,
  },
  icon: {
    marginRight: 5,
  },
  label: {
    position: 'absolute',
    backgroundColor: 'white',
    left: 22,
    top: 8,
    zIndex: 999,
    paddingHorizontal: 8,
    fontSize: 14,
  },
  placeholderStyle: {
    fontSize: 16,
  },
  selectedTextStyle: {
    fontSize: 16,
  },
  iconStyle: {
    width: 20,
    height: 20,
  },
  inputSearchStyle: {
    height: 40,
    fontSize: 16,
  },
});
