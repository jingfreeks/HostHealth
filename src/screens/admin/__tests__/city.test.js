import React from 'react'
import {renderWithProviders} from '@/utils/testframeworknew';
import {City} from '../city';
describe('City admin screen', () => {
  it('Should work as expected to get snapshot', () => {
    const all = renderWithProviders(<City />);
    expect(all.toJSON()).toMatchSnapshot();
  });
});
