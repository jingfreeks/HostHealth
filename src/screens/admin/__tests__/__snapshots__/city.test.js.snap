// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City Form admin screen Should work as expected to get snapshot 1`] = `
<View
  style={
    [
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "marginLeft": 39.23529411764706,
        "marginRight": 39.23529411764706,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "marginBottom": 39.23529411764706,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#0e3f6c",
            "fontFamily": "Avenir",
            "fontSize": 35.311764705882354,
            "fontWeight": "500",
          },
          {
            "fontSize": 30,
            "fontWeight": "900",
          },
        ]
      }
    >
      City Form Information
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": 20,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "marginLeft": 20,
          },
        ]
      }
    >
      <View
        style={
          {
            "backgroundColor": "transparent",
            "borderRadius": 125,
            "height": 250,
            "width": 250,
          }
        }
        theme={
          {
            "colors": {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": {
                "android": {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            {
              "backgroundColor": "transparent",
              "borderRadius": 125,
              "flex": 1,
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <Image
            borderRadius={125}
            onLoad={[Function]}
            source={
              {
                "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/ladylexy/128.jpg",
              }
            }
            style={
              {
                "bottom": 0,
                "flex": 1,
                "height": undefined,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": undefined,
              }
            }
            testID="RNE__Image"
            theme={
              {
                "colors": {
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": {
                    "android": {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
              }
            }
            transition={true}
            transitionDuration={360}
          />
          <View
            accessibilityElementsHidden={true}
            collapsable={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            style={
              {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#bdbdbd",
                  "flex": 1,
                  "height": undefined,
                  "justifyContent": "center",
                  "width": undefined,
                }
              }
              testID="RNE__Image__placeholder"
            />
          </View>
          <View
            style={
              {
                "flex": 1,
                "height": undefined,
                "width": undefined,
              }
            }
            testID="RNE__Image__children__container"
          />
        </View>
      </View>
      <View
        accessibilityLabel="ProfileAvatarUploadImageTestId"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "flex-end",
            "justifyContent": "flex-end",
            "opacity": 1,
          }
        }
        testID="ProfileAvatarUploadImageTestId"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#C2D6DC",
                "fontSize": 25,
              },
              undefined,
              {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          î«œ
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "marginBottom": 9.808823529411764,
          "marginTop": 39.23529411764706,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "marginLeft": 5,
          "marginRight": 5,
          "paddingBottom": 5,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 5,
        },
      ]
    }
  >
    <View
      border={10}
      borderc="#99E1E1"
      borderw={2}
      style={
        [
          {
            "borderColor": "#99E1E1",
            "borderRadius": 10,
            "borderWidth": 2,
          },
        ]
      }
    >
      <View
        accessibilityLabel="AtomsButtonTestId"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#d6f3f3",
            "borderRadius": 10,
            "opacity": 1,
            "paddingBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 15,
          }
        }
        testID="AtomsButtonTestId"
      >
        <Text
          style={
            [
              {
                "color": "#03606A",
                "fontFamily": "Avenir",
                "fontSize": 15,
                "fontWeight": "500",
              },
            ]
          }
        >
          Save
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`City admin screen Should work as expected to get snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "center",
      "flex": 1,
    }
  }
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={false}
    />
  }
>
  <RCTRefreshControl />
  <View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <RCTSafeAreaView
        style={
          [
            {},
          ]
        }
      >
        <View
          alignType="center"
          style={
            [
              {
                "alignItems": "center",
                "alignSelf": "center",
                "flexDirection": "column",
              },
            ]
          }
        >
          <Image
            resizeMode="contain"
            size={150}
            source={
              {
                "testUri": "../../../src/assets/pcitiesempty.png",
              }
            }
            style={
              [
                {
                  "height": 294.2647058823529,
                  "width": 294.2647058823529,
                },
              ]
            }
          />
          <Text
            style={
              [
                {
                  "color": "#000000",
                  "fontFamily": "Avenir",
                  "fontSize": 16,
                  "fontWeight": "400",
                },
              ]
            }
          >
            No cities matches your preferred locations
          </Text>
        </View>
      </RCTSafeAreaView>
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "borderRadius": 28,
            "bottom": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 16,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "transform": [
              {
                "scale": 1,
              },
            ],
          }
        }
      >
        <View
          style={
            [
              {
                "borderRadius": 3,
                "overflow": "hidden",
              },
              {
                "borderRadius": 3,
              },
              {
                "borderRadius": 28,
                "elevation": 4,
              },
              false,
            ]
          }
        >
          <View
            accessibilityLabel="CityCreateButtonTestId"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": false,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
            testID="CityCreateButtonTestId"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#ca71eb",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "height": 48,
                  "justifyContent": "center",
                  "padding": 8,
                  "paddingHorizontal": 16,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#ffffff",
                    "fontSize": 18,
                    "marginHorizontal": 8,
                    "paddingVertical": 1,
                    "textAlign": "center",
                  }
                }
              >
                Create
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
